presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -163,13 +166,14 @@ public static ParquetPageSource createParquetPageSource(
             MessageType fileSchema = fileMetaData.getSchema();
             dataSource = buildHdfsParquetDataSource(inputStream, path, fileSize, stats);
 
-            List<org.apache.parquet.schema.Type> fields = columns.stream()
+            List<Optional<org.apache.parquet.schema.Type>> parquetFields = columns.stream()
                     .filter(column -> column.getColumnType() == REGULAR)
                     .map(column -> getParquetType(column, fileSchema, useParquetColumnNames))
-                    .filter(Objects::nonNull)
-                    .collect(toList());
+                    .map(Optional::ofNullable)
+                    .collect(toImmutableList());
 
-            MessageType requestedSchema = new MessageType(fileSchema.getName(), fields);
+            MessageType requestedSchema = new MessageType(fileSchema.getName(),
```

Comment:
Let's format this so it's easier to see the stream operations
```java
MessageType requestedSchema = new MessageType(
       fileSchema.getName(),
       parquetFields.stream()
                .filter(Optional::isPresent)
                .map(Optional::get)
                .collect(toImmutableList()));
```
Though we can actually simplify this using Guava
```java
MessageType requestedSchema = new MessageType(
       fileSchema.getName(),
       parquetFields.stream()
                .flatMap(Streams::stream)
                .collect(toImmutableList()));
```
In Java 9, this is replaced with `Optional::stream`.